### PR TITLE
Add new design and instruction prompt

### PR DESCRIPTION
**Pull request by niklaspaasonen**
### This PR will:
- Spruce up the styling and layout
- Add a much more detailed (and seemingly better-performing) initial system prompt 
- Delete old example data and add the latest version of the HTML file (chapter 4 only)
- Swap out the old prompt examples for ones that the model handles well

**Note:** in order to use the `gtp-4` model, it seems it was actually the env variable `AZURE_OPENAI_CHATGPT_MODEL` that had to be changed.

### CODE REVIEW

File: scripts/prepdocslib/listfilestrategy.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ async def _list_paths(self, path_pattern: str) -> AsyncGenerator[str, None]:
     async def list(self) -> AsyncGenerator[File, None]:
         async for path in self.list_paths():
             if not self.check_md5(path):
-                yield File(content=open(path, mode="rb"))
+                yield File(content=open(path, mode="r"))
```
Comment (nPaasonen) [inline scripts/prepdocslib/listfilestrategy.py:78]: @nj-kore do you have an opinion about this

File: scripts/prepdocs.py [hunk 2]
```diff
@@ -36,6 +36,26 @@ def setup_file_strategy(credential: AsyncTokenCredential, args: Any) -> FileStra
...
+        # formrecognizer_creds: Union[AsyncTokenCredential, AzureKeyCredential] = (
+        #     credential if is_key_empty(args.formrecognizerkey) else AzureKeyCredential(args.formrecognizerkey)
+        # )
+        # pdf_parser = DocumentAnalysisPdfParser(
+        #     endpoint=f"https://{args.formrecognizerservice}.cognitiveservices.azure.com/",
+        #     credential=formrecognizer_creds,
+        #     verbose=args.verbose,
+        # )
+
+
```
Comment (nj-kore) [inline scripts/prepdocs.py]: This should be removed.

File: scripts/prepdocslib/contentparsers.py [hunk 3]
```diff
@@ -151,3 +151,13 @@ def table_to_html(cls, table: DocumentTable):
...
+
+
+class TextParser:
+    """
+    Abstract parser that parses pages into text
+    """
+
+    async def parse(self, content: IO) -> AsyncGenerator[Page, None]:
+        text = content.read()
+        yield Page(page_num=0, offset=0, text=text)
```
Comment (nj-kore) [inline scripts/prepdocslib/contentparsers.py]: This should be removed. We already have a TextParser that I defined higher up in the file.

**Review by nj-kore — APPROVED** [review]
Great job! Just remove the two things I commented on :)

**Pull request merged**